### PR TITLE
Add architecture agnostic namespace to be used in Zephyr

### LINKED ISSUE #52: Update the header prefixes to be architecture agnostic
I have a need for a similar math library in Zephyr which needs to work for and ideally be optimized for additional architectures beyond ARM. I was wondering if it would be possible to rename the headers and replace the `arm_` prefix with `cmsis_`? the `arm_` prefixed headers can still exist and simply include their generic counterparts. This would allow us to provide Zephyr implementations for non arm optimized math logic.

Reference https://github.com/zephyrproject-rtos/zephyr/pull/50012

### PR DESCRIPTION
**Pull request by yperess**
Fixes #52

### CODE REVIEW

**Comment by yperess** [comment]
@christophe0606 I've uploaded this WIP PR, it's a very very small start. I just wanted to check and make sure this is what you had in mind before I do all of the headers

File: Include/cmsis_dsp_math_types.h [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+#include "arm_math_types.h"
+
+#define CMSIS_MATH_SUCCESS ARM_MATH_SUCCESS
+#define CMSIS_MATH_ARGUMENT_ERROR ARM_MATH_ARGUMENT_ERROR
+#define CMSIS_MATH_LENGTH_ERROR ARM_MATH_LENGTH_ERROR
+#define CMSIS_MATH_SIZE_MISMATCH ARM_MATH_SIZE_MISMATH
+#define CMSIS_MATH_MANIF ARM_MATH_MANIF
+#define CMSIS_MATH_TEST_FAILURE ARM_MATH_TEST_FAILURE
+#define CMSIS_MATH_DECOMPOSITION_FAILURE ARM_MATH_DECOMPOSITION_FAILURE
```
Comment (christophe0606) [inline Include/cmsis_dsp_math_types.h:16]: I may perhaps add those to `arm_math_types.h`. A developer can (for instance) only include `dsp/basic_math_functions.h` if there is no need to include all of the headers. For instance, if the application is only using a subset of the functions.

This `basic_math_functions.h` is including `arm_math_types.h` so the developer would only have access to the `ARM_MATH_` error codes without those defines.

There is no other public symbol that would need to be defined.

`basic_math_functions.h`  is including

```
#include "arm_math_types.h"
#include "arm_math_memory.h"

#include "dsp/none.h"
#include "dsp/utils.h"
```

The `arm_math_types.h` doesn't have any `arm_` prefix except for the error code.
The `arm_math_memory.h` is providing some memory functions with no `arm_` prefix.

`none.h` is used to provide C implementation of some intrinsics when they are missing. No `arm_` prefix.

`utils.h` is containing some functions with `arm_` prefix and so may also require a few define inside. For instance

`#define cmsis_dsp_div_int64_to_int32 arm_div_int64_to_int32`

**Review by christophe0606 — CHANGES_REQUESTED** [review]
I realize that in the `Include` folder, lot of files should instead be in the `PrivateInclude` folder since they will never be used by the user of the library but are only used by the implementation.
I am creating an issue to track this and will do the change at a later time.

The public includes for which a `cmsis_dsp_` wrapper is needed are:
```
arm_math_types.h 
arm_math_types_f16.h

arm_math.h
arm_math_f16.h

arm_const_structs.h
arm_const_structs_f16.h
```
`arm_const_structs.h` is used for old style initialization of the FFT and still used in lot of existing code in the ecosystem.

**Comment by yperess** [comment]
Thank you for the feedback, I'll update on Thursday after the holiday.

**Pull request closed without merging**